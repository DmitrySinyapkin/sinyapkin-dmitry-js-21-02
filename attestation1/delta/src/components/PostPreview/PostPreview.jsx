import React from "react";
import './PostPreview.css';
import { Avatar } from 'antd';
import HelperWrapper from "../../wrappers/HelperWrapper/HelperWrapper";
import { Link } from "react-router-dom";

const PostPreview = (props) => {
    return (
        <div className={`post__container ${props.darkTheme && 'post_dark'}`}>
            <Avatar className={`post__avatar ${props.hidden && "hidden"}`} shape="circle" src={props.avatar} />
            <HelperWrapper text={props.id}>
                <div className={`post__name ${props.hidden && "hidden"}`}><Link to={`/users/${props.id}`}>{props.title ? `${props.title} ${props.firstName} ${props.lastName}` : `${props.firstName} ${props.lastName}`}</Link></div>
            </HelperWrapper>
            <div className={`post__date ${props.hidden && "hidden"}`}>{props.date}</div>
            <figure className="post__picture">
                <img src={props.picture} alt="Load error" />
            </figure>
            <div className="post__text-preview">{props.text}</div>
        </div>
    );
}

export default PostPreview;
