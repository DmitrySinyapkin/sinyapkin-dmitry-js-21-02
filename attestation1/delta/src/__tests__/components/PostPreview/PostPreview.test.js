import React from "react";
import { shallow } from 'enzyme';
import PostPreview from '../../../components/PostPreview/PostPreview';

describe('PostPreview component', () => {
    test('should render component', () => {
        const wrapper = shallow(<PostPreview />);
        expect(wrapper.find('div.post__container')).toHaveLength(1);
    });

    test('snapshot test', () => {
        const wrapper = shallow(<PostPreview />);
        expect(wrapper).toMatchSnapshot();
      });

    test('should render data', () => {
        const title = 'mr';
        const firstName = 'James';
        const lastName = 'Bond';
        const date = '12.06.2020';
        const text = 'Blah-blah-blah';
        const picture = 'pictureUrl';
        const wrapper = shallow(<PostPreview
           title={title}
           firstName={firstName} 
           lastName={lastName}
           date={date}
           text={text}
           picture={picture}
        />);
        expect(wrapper.find('div.post__name').first().text()).toBe(`${title} ${firstName} ${lastName}`);
        expect(wrapper.find('div.post__date').text()).toBe(date);
        expect(wrapper.find('div.post__text-preview').text()).toBe(text);
        expect(wrapper.find('figure.post__picture').find('img').prop('src')).toBe(picture);
    });
});
