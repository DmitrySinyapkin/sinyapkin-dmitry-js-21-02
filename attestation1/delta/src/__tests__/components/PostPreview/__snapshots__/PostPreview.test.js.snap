// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreview component snapshot test 1`] = `
<div
  className="post__container undefined"
>
  <Avatar
    className="post__avatar undefined"
    shape="circle"
    size="default"
  />
  <HelperWrapper>
    <div
      className="post__name undefined"
    >
      <Link
        to="/users/undefined"
      >
        undefined undefined
      </Link>
    </div>
  </HelperWrapper>
  <div
    className="post__date undefined"
  />
  <figure
    className="post__picture"
  >
    <img
      alt="Load error"
    />
  </figure>
  <div
    className="post__text-preview"
  />
</div>
`;
